src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java

```diff
@@ -649,47 +662,127 @@
   }
 
   @Override
-  public DataFetcher<Iterable<BikePark>> bikeParks() {
+  public DataFetcher<Iterable<VehicleParking>> bikeParks() {
     return environment -> {
-      VehicleRentalStationService vehicleRentalStationService = getRoutingService(environment)
-          .getVehicleRentalStationService();
+      VehicleParkingService vehicleParkingService = getRoutingService(environment)
+          .getVehicleParkingService();
 
-      if (vehicleRentalStationService == null) { return null; }
+      if (vehicleParkingService == null) { return null; }
 
-      return vehicleRentalStationService.getBikeParks();
+      return vehicleParkingService.getBikeParks().collect(Collectors.toList());
     };
   }
 
   @Override
-  public DataFetcher<BikePark> bikePark() {
+  public DataFetcher<VehicleParking> bikePark() {
     return environment -> {
       var args = new LegacyGraphQLTypes.LegacyGraphQLQueryTypeBikeParkArgs(environment.getArguments());
 
-      VehicleRentalStationService vehicleRentalStationService = getRoutingService(environment)
-          .getVehicleRentalStationService();
+      VehicleParkingService vehicleParkingService = getRoutingService(environment)
+          .getVehicleParkingService();
 
-      if (vehicleRentalStationService == null) { return null; }
+      if (vehicleParkingService == null) { return null; }
 
-      return vehicleRentalStationService
+      return vehicleParkingService
               .getBikeParks()
-              .stream()
-              .filter(bikepark -> bikepark.id.equals(args.getLegacyGraphQLId()))
+              .filter(bikePark -> bikePark.getId().getId().equals(args.getLegacyGraphQLId()))
               .findAny()
               .orElse(null);
     };
   }
 
-  //TODO
   @Override
-  public DataFetcher<Iterable<Object>> carParks() {
-    return environment -> Collections.EMPTY_LIST;
+  public DataFetcher<Iterable<VehicleParking>> carParks() {
+    return environment -> {
+      VehicleParkingService vehicleParkingService = getRoutingService(environment)
+          .getVehicleParkingService();
+
+      if (vehicleParkingService == null) { return null; }
+
+      var args = new LegacyGraphQLTypes.LegacyGraphQLQueryTypeCarParksArgs(
+          environment.getArguments());
+
+      if (args.getLegacyGraphQLIds() != null) {
+        var idList = ((List<String>) args.getLegacyGraphQLIds());
+
+        if (!idList.isEmpty()) {
+          Map<String, VehicleParking> carParkMap = vehicleParkingService.getCarParks()
+                  .collect(Collectors.toMap(station -> station.getId().toString(), station -> station));
```

Comment:
Forgot the carParks query also had ids parameter so this needs to be changed as well.